web/app.js

```diff
@@ -687,11 +686,11 @@ var PDFViewerApplication = {
     var loadingTask = pdfjsLib.getDocument(parameters);
     this.pdfLoadingTask = loadingTask;
 
-    loadingTask.onPassword = function passwordNeeded(updatePassword, reason) {
-      PasswordPrompt.updatePassword = updatePassword;
-      PasswordPrompt.reason = reason;
-      PasswordPrompt.open();
-    };
+    loadingTask.onPassword = function passwordNeeded(callback, reason) {
+      this.passwordPrompt.setCallback(callback);
+      this.passwordPrompt.setReason(reason);
```

Comment:
I removed the unnecessary `bind` in the new commit.

I thought it would be more clear to use one setter per member variable. Furthermore, if we were to combine the two, I wouldn't know a good method name for it. If you think it's necessary to combine the two, do you have a suggestion for a name?